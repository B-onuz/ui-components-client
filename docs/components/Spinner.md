# Spinner (Component)

Componente para tratar estado de loading.

## Como utilizar

```js
import Spinner from 'ui-components/lib/components/Spinner'
```

```jsx
<Spinner />
```

## Variações

### Variant

```jsx
<Spinner color={'primary'} />
```

Escala no tema: `theme.colors`

## Styled-system props

- space https://styled-system.com/table#space
- layout https://styled-system.com/table#layout
