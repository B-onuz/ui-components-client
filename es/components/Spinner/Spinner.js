"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Spinner/Spinner.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  display: inline-flex;\n  &,\n  &:after {\n    border-radius: 50%;\n    width: 2em;\n    height: 2em;\n  }\n  font-size: 10px;\n  vertical-align: middle;\n  position: relative;\n  text-indent: -9999em;\n  border-top: 2px solid ", ";\n  border-right: 2px solid ", ";\n  border-bottom: 2px solid ", ";\n  border-left: 2px solid #ffffff;\n  -webkit-transform: translateZ(0);\n  -ms-transform: translateZ(0);\n  transform: translateZ(0);\n  -webkit-animation: load8 1.1s infinite linear;\n  animation: load8 0.5s infinite linear;\n  ", "\n  @keyframes load8 {\n    0% {\n      -webkit-transform: rotate(0deg);\n      transform: rotate(0deg);\n    }\n    100% {\n      -webkit-transform: rotate(360deg);\n      transform: rotate(360deg);\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledSpinner = _styledComponents["default"].div(_templateObject(), function (_ref) {
  var theme = _ref.theme,
      color = _ref.color;
  return "" + theme.colors[color];
}, function (_ref2) {
  var theme = _ref2.theme,
      color = _ref2.color;
  return "" + theme.colors[color];
}, function (_ref3) {
  var theme = _ref3.theme,
      color = _ref3.color;
  return "" + theme.colors[color];
}, (0, _styledSystem.compose)(_styledSystem.space, _styledSystem.layout));

var Spinner = function Spinner(_ref4) {
  var children = _ref4.children,
      rest = _objectWithoutPropertiesLoose(_ref4, ["children"]);

  return /*#__PURE__*/_react["default"].createElement(StyledSpinner, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 40,
      columnNumber: 10
    }
  }));
};

var _default = Spinner;
exports["default"] = _default;