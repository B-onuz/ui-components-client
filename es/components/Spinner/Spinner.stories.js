'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.SpinnerExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Spinner = _interopRequireDefault(require('./Spinner'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Spinner/Spinner.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'Spinner',
  component: _Spinner['default'],
}
exports['default'] = _default

var SpinnerExample = function SpinnerExample() {
  return /*#__PURE__*/ _react['default'].createElement(_Spinner['default'], {
    color: 'primary',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 9,
      columnNumber: 37,
    },
  })
}

exports.SpinnerExample = SpinnerExample
