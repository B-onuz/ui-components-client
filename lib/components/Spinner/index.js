"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _Spinner = _interopRequireDefault(require("./Spinner"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _Spinner["default"];
exports["default"] = _default;
module.exports = exports.default;