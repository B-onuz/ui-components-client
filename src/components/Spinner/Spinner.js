import React from 'react'
import styled from 'styled-components'
import { space, layout, compose } from 'styled-system'

const StyledSpinner = styled.div`
  display: inline-flex;
  &,
  &:after {
    border-radius: 50%;
    width: 2em;
    height: 2em;
  }
  font-size: 10px;
  vertical-align: middle;
  position: relative;
  text-indent: -9999em;
  border-top: 2px solid ${({ theme, color }) => `${theme.colors[color]}`};
  border-right: 2px solid ${({ theme, color }) => `${theme.colors[color]}`};
  border-bottom: 2px solid ${({ theme, color }) => `${theme.colors[color]}`};
  border-left: 2px solid #ffffff;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: load8 1.1s infinite linear;
  animation: load8 0.5s infinite linear;
  ${compose(space, layout)}
  @keyframes load8 {
    0% {
      -webkit-transform: rotate(0deg);
      transform: rotate(0deg);
    }
    100% {
      -webkit-transform: rotate(360deg);
      transform: rotate(360deg);
    }
  }
`

const Spinner = ({ children, ...rest }) => {
  return <StyledSpinner {...rest} />
}

export default Spinner
