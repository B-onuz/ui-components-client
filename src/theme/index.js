const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
