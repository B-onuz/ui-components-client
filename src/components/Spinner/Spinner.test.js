import React from 'react'
import Spinner from './Spinner.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Spinner changes the class when hovered', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Spinner />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
