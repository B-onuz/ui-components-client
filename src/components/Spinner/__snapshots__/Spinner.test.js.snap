// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner changes the class when hovered 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 10px;
  vertical-align: middle;
  position: relative;
  text-indent: -9999em;
  border-top: 2px solid undefined;
  border-right: 2px solid undefined;
  border-bottom: 2px solid undefined;
  border-left: 2px solid #ffffff;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: load8 1.1s infinite linear;
  -webkit-animation: load8 0.5s infinite linear;
  animation: load8 0.5s infinite linear;
}

.c0,
.c0:after {
  border-radius: 50%;
  width: 2em;
  height: 2em;
}

<div
  className="c0"
/>
`;
